core/src/test/java/org/apache/spark/SparkThrowableSuite.java

```diff
@@ -0,0 +1,150 @@
+package org.apache.spark;
+
+import java.io.*;
+import java.nio.charset.Charset;
+import java.util.Map;
+import java.util.HashSet;
+import java.util.IllegalFormatException;
+import java.util.SortedMap;
+
+import com.fasterxml.jackson.core.JsonParser.Feature;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import org.apache.commons.io.IOUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+import org.apache.spark.SparkThrowable.SparkThrowableHelper;
+
+public class SparkThrowableSuite implements Serializable {
```

Comment:
I think the test in Scala is good enough, which also covers the most common case: we usualy extend `SparkThrowable` in the scala code (spark codebase is mostly scala).

Reply:
Done - I also added a check on the Java side